envoy/api/v2/discovery.proto

```diff
@@ -36,6 +36,10 @@ message DiscoveryRequest {
   // explicitly enumerated in resource_names.
   repeated string resource_names = 3;
 
+  // When true the list of resource_names is added to the set of tracked resources
+  // by this node.
+  bool incremental = 4;
```

Comment:
Would you want a way to remove tracked resources incrementally?

We should be clear that when the stream is dropped, this state disappears.

Reply:
I think incremental removal is handled below? More generally though, we need to think through reconnection. I think we need to build something into the incremental protocol where the first response on a stream is the full dump, and then it switches to incremental, or something like that. Otherwise in incremental mode Envoy will have to tell the management server everything it knows about when in incremental mode?

Reply:
The semantics of incremental are a bit different here. In this particular field, incremental refers to changing the list of tracked resources (or resource hints) maintained by the management server for a given xDS resource. Today, for CDS, this is the empty set, but in the future this will be the set of clusters that have been lazily loaded and the management server needs to know to update when changes come along at its end.

Reply:
I see, OK. Then yes we definitely need incremental removal also, in addition to the discussion about reconnection.